src/main/java/org/passay/PasswordGenerator.java

```diff
@@ -57,10 +57,12 @@ public String generatePassword(final int length, final List<CharacterRule> rules
 
     final CharBuffer buffer = CharBuffer.allocate(length);
     if (rules != null) {
-      for (CharacterRule rule : rules) {
+      rules.stream().map((rule) -> {
```

Comment:
I'm all for the streaming API if it improves clarity or reduces code. I don't see that it does either here.